package.json

```
@@ -935,12 +935,12 @@
         "visible": true
       },
       {
-        "permissionName": "mod-settings.global.read.ui-inventory.display-settings",
+        "permissionName": "mod-settings.global.ui-inventory.display-settings.view",
         "displayName": "Settings (Inventory): read the settings in the 'Display settings' section",
         "visible": false
```

@Dmytro-Melnyshyn maybe it can be refactored this way?

```suggestion
        "permissionName": "ui-inventory.display-settings.view",
        "displayName": "Settings (Inventory): read the settings in the 'Display settings' section",
        "subPermissions": ["mod-settings.global.read.ui-inventory.display-settings"],
        "visible": false
```

---

Hey @mariia-aloshyna, `mod-settings` is a replacement for `mod-configuration` due to security issues with the latter. It allows storing data on the BE side without disturbing the BE team. It is useful for flags. User must have `mod-settings.global.write.{scope}` permission to `POST/PUT/DELETE` to this storage, and have `mod-settings.users.read.{scope}` permission for `GET` requests. Only `scope` part can be made up.

To be able to use that storage the UI needs to create the _BE_ `mod-settings.global.read/write.{scope}` permissions, which is done as follows:

```
      {
        "permissionName": "mod-settings.global.read.ui-inventory.display-settings",
        "displayName": "Settings (Inventory): read the settings in the 'Display settings' section",
        "visible": false
      },
      {
        "permissionName": "mod-settings.global.write.ui-inventory.display-settings",
        "displayName": "Settings (Inventory): write the settings in the 'Display settings' section",
        "visible": false
      },
 ```
      
And then these permissions are used in the new `ui-*` permission:

```
      {
        "permissionName": "ui-inventory.settings.displaySettings",
        "displayName": "Settings (Inventory): Can view and edit general settings",
        "subPermissions": [
          "settings.inventory.enabled",
          "mod-settings.entries.collection.get",
          "mod-settings.entries.item.post",
          "mod-settings.entries.item.put",
          "mod-settings.global.read.ui-inventory.display-settings",
          "mod-settings.global.write.ui-inventory.display-settings"
        ],
        "visible": true
      }
```

I'm okay with changing the UI permission name - `ui-inventory.settings.displaySettings`. But I'm not sure how it is written on the BE side, and whether changing BE (`mod-settings.global.read/write.{scope}`) permission names breaks functionality or not. I wouldn't change them until the requirements are rewritten in the `mod-settings` [documentation](https://github.com/folio-org/mod-settings?tab=readme-ov-file#introduction).

---

The interface has been added to the `stripes-inventory-components` repository.